common/json.c

```diff
@@ -155,6 +155,30 @@ bool json_to_bitcoin_amount(const char *buffer, const jsmntok_t *tok,
 	return true;
 }
 
+u8 *json_tok_bin_from_hex(const tal_t *ctx, const char *buffer, const jsmntok_t *tok)
+{
+	u8 *result;
+	size_t hexlen, rawlen;
+	hexlen = tok->end - tok->start;
+	rawlen = hex_data_size(hexlen);
+
+	result = tal_arr(ctx, u8, rawlen);
+	if (!hex_decode(buffer + tok->start, hexlen, result, rawlen))
+		return tal_free(result);
+
+	return result;
+}
+
+bool json_to_preimage(const char *buffer, const jsmntok_t *tok, struct preimage *preimage)
+{
+	size_t hexlen = tok->end - tok->start;
+
+	if (hexlen != sizeof(preimage->r))
```

Comment:
Pretty sure hex_decode already does this check?

Reply:
:+1: will drop duplicate check :-)